src/cephadm/cephadm

```diff
@@ -4608,6 +4609,526 @@ def command_install():
+        """Determine OS version"""
+        raw_info = read_file(['/etc/os-release'])
+        os_release = raw_info.splitlines()
+        rel_str = 'Unknown'
+        rel_dict = dict()
+
+        for line in os_release:
+            if "=" in line:
+                var_name, var_value = line.split('=')
+                rel_dict[var_name] = var_value.strip('"')
+
+        # Would normally use PRETTY_NAME, but NAME and VERSION are more
+        # consistent
+        if all(_v in rel_dict for _v in ["NAME", "VERSION"]):
+            rel_str = "{} {}".format(rel_dict['NAME'], rel_dict['VERSION'])
+        return rel_str
+
+    @property
+    def hostname(self):
+        # type: () -> str
+        """Return the hostname"""
+        return platform.node()
+
+    @property
+    def subscribed(self):
+        # type: () -> str
+        """Highlevel check to see if the host is subscribed to receive updates/support"""
+        def _red_hat():
+            # type: () -> str
+            # RHEL 7 and RHEL 8
+            entitlements_dir = '/etc/pki/entitlement'
+            if os.path.exists(entitlements_dir):
+                pems = glob('{}/*.pem'.format(entitlements_dir))
+                if len(pems) >= 2:
+                    return "Yes"
+
+            return "No"
+
+        os_name = self.operating_system
+        if os_name.upper().startswith("RED HAT"):
+            return _red_hat()
+
+        return "Unknown"
+
+    @property
+    def hdd_count(self):
+        # type: () -> int
+        """Return a count of HDDs (spinners)"""
+        return len(self._get_devs_by_type(rota='1'))
+
+    def _get_capacity(self, dev):
+        # type: (str) -> int
+        """Determine the size of a given device"""
+        size_path = os.path.join('/sys/block', dev, 'size')
+        size_blocks = int(read_file([size_path]))
+        blk_path = os.path.join('/sys/block', dev, 'queue', 'logical_block_size')
+        blk_count = int(read_file([blk_path]))
+        return size_blocks * blk_count
+
+    def _get_capacity_by_type(self, rota='0'):
+        # type: (str) -> int
+        """Return the total capacity of a category of device (flash or hdd)"""
+        devs = self._get_devs_by_type(rota=rota)
+        capacity = 0
+        for dev in devs:
+            capacity += self._get_capacity(dev)
+        return capacity
+
+    def _dev_list(self, dev_list):
+        # type: (List[str]) -> List[str]
+        """Return a 'pretty' name list for each device in the `dev_list`"""
+        disk_list = list()
+
+        for dev in dev_list:
+            disk_model = read_file(['/sys/block/{}/device/model'.format(dev)]).strip()
+            vendor = read_file(['/sys/block/{}/device/vendor'.format(dev)]).strip()
+            disk_vendor = HostFacts._disk_vendor_workarounds.get(vendor, vendor)
+            disk_size_bytes = self._get_capacity(dev)
+            disk_list.append("{} {} ({})".format(disk_vendor, disk_model, bytes_to_human(disk_size_bytes)))
```

Comment:
I'm getting

```
  "hdd_list": [
    "Unknown Unknown (4.2MB)",
    "Unknown Unknown (101.7MB)",
    "Unknown Unknown (58.0MB)",
    "Unknown Unknown (10.6MB)",
    "Unknown Unknown (130.9MB)",
    "Unknown Unknown (130.0MB)",
    "Unknown Unknown (57.6MB)",
    "Unknown Unknown (10.3MB)",
    "Unknown Unknown (101.3MB)",
    "Unknown Unknown (3.0MB)",
    "ATA ST2000LM007-1R81 (2.0TB)",
    "Unknown Unknown (0.0)",
    "Unknown Unknown (130.4MB)",
    "Unknown Unknown (10.6MB)",
    "Unknown Unknown (10.4MB)",
    "Unknown Unknown (4.2MB)"
  ],
```

I'm just wondering how useful that data is. 

* Would it make sense to expose some more information? 
* Would it make sense to expose this as objects?

```json
"hdd_list": [
  {
    "vendor": "Unknown",
    "model": "Unknown"
    "disk_size": ...
  }
]
```

this way the structure is extensible without intoducing potential backward incompatible changes

Reply:
what on earth are those drives :) 
Gather facts is supposed to be quick and easy - I'll patch to use a dict per device for extensibility and include a description field that shows what this current does. I don't want to include objects - my aim is keep it simple - gather-facts is supposed to be a quick scan of the machine - anything more complicated than that needs ceph-volume etc etc

Reply:
fyi, the teuthology machines return something similar:

```json
"hdd_list": [
  "Unknown Unknown (0.0)",
  "Unknown Unknown (0.0)",
  "Unknown Unknown (0.0)",
  "Unknown Unknown (0.0)",
  "Unknown Unknown (0.0)",
  "Unknown Unknown (0.0)",
  "ATA ST1000NM0033-9ZM (1.0TB)",
  "Unknown Unknown (0.0)",
  "Unknown Unknown (0.0)"
]
```

http://qa-proxy.ceph.com/teuthology/swagner-2020-08-18_11:50:15-rados:cephadm-wip-swagner3-testing-2020-08-18-1033-distro-basic-smithi/5356265/teuthology.log

Reply:
> I don't want to include objects - my aim is keep it simple

This will make it hard for the dashboard to display vendor and model etc in a table or something. @ceph/dashboard folks: Are you ok with this?

Reply:
If it's easy to achieve it would be useful to have that info.

Reply:
gather-facts returns JSON - why would that make it hard for the dashboard to consume?

Latest patch breaks down the disk metadata, so it is a list of dicts - as follows;
```
      "description": "Virtio Block Device Unknown (42.9GB)",
      "dev_name": "vda",
      "disk_size_bytes": 42949672960,
      "model": "Unknown",
      "rev": "Unknown",
      "vendor": "Virtio Block Device",
      "wwid": "Unknown"
    }
```
@alfonsomthd maybe - but it would be nice to get this in place first!